app/ControllerFunctions/Install/RequirementsChecker.php

```diff
@@ -28,21 +28,28 @@ public function check(array $requirements)
 					foreach ($requirements[$type] as $requirement) {
 						$results['requirements'][$type][$requirement] = true;
 						if (!extension_loaded($requirement)) {
+							// @codeCoverageIgnoreStart
 							$results['requirements'][$type][$requirement]
 								= false;
 							$results['errors'] = true;
+							// @codeCoverageIgnoreEnd
 						}
 					}
 
 					if ($this->checkExec()) {
 						$results['requirements'][$type]['Php exec() available'] = true;
+					} else {
+						// @codeCoverageIgnoreStart
+						$results['requirements'][$type]['Php exec() not available (optional)'] = false;
```

Comment:
This is familiar :P But I'm confused that they're different strings/indexes *and* different values? i.e. `{ yes = true; } else { no = false; }`

Reply:
Ah. Because it just lists through and shows `true` as a tick and `false` as a cross. I remember.